### DISCUSSION TITLE
Answered: looking for a  small typescript project example

### DISCUSSION THREAD

**Discussion by markfirmware** [alpinejs/alpine#1005] [1. Help]
.

**Comment by markfirmware** [comment]
Any ideas?

**Comment by markfirmware** [comment]
I tried copying https://alpinejs.codewithhugo.com/x-data-access/ into codepen at https://codepen.io/markfirmware/pen/OJRrjpo

**Comment by markfirmware** [comment]
https://2020.stateofjs.com/en-US/technologies/javascript-flavors/

Does typescript conflict with the rugged nature of alpinejs? I would be just as happy to dismiss ts with cause as adopt it.

**Reply by KevinBatdorf** [reply]
I have very little experience with TypeScript but it has to be compiled so you can't use it in your HTML, so you would have to extract all your functions out to take advantage of TypeScript's features. 

I think you can use it... or not. Completely up to you.

**Reply by markfirmware** [reply]
Ok, I was just starting to think that the compile step might provide some friction. Thanks.

**Reply by KevinBatdorf** [reply]
I think the benefits to TypeScript and your decision to use it should be independent of your using Alpine. I personally prefer my JS to be dynamically typed, but that's just a personal preference.

**Reply by markfirmware** [reply]
Right. I can see that putting ts into say @click would require some kind of surgical compilation and might not even be well-defined without defining semantics. Enough said.

**Reply by markfirmware** [reply]
The discussions will only let me mark my own comments as an answer. I will just update the title.
